File: Sample Code/JavaScript/Random Numbers/Random Numbers/random-lcg.js
// ------------------------------------------------------------------
//
// Linear Congruential Generator (LCG)
//
// Reference: https://en.wikipedia.org/wiki/Linear_congruential_generator
//
// ------------------------------------------------------------------
let Random = (function() {
    'use strict';
    // Following C++ 11: m = 2^32 - 1, a = 48271, c = 0
    let m = 4294967295;
    let a = 47271;
    let c = 0;
    let z = 3;  // The seed

    // Use these to demonstrate the period
    // let m = 25;
    // let a = 11;
    // let c = 17;

    function setSeed(seed) {
        z = seed
    }

    function next() {
        z = (a * z + c) % m;
        return z;
    }

    function nextDouble() {
        return next() / m;
    }

    return {
        setSeed, setSeed,
        next: next,
        nextDouble : nextDouble
    };

}());


for (i = 0; i < 100; i++) {
    //console.log(`${i}: ${Random.nextDouble()}`);  // For demonstrating the period
    console.log(Random.nextDouble());
}

File: Sample Code/MonoGame/RandomNumbers/RandomNumbers/RandomNumbers/RandomLCG.cs
namespace Random
{
    public class RandomLCG
    {
        // Following C++ 11: m = 2^32 - 1, a = 48271, c = 0
        private long m = 4294967295;
        private long a = 47271;
        private long c = 0;
        private long z = 3;

        // Use these to demonstrate the period
        // private long m = 25;
        // private long a = 11;
        // private long c = 17;
        public RandomLCG(long seed = 3)
        {
            this.z = seed;
        }

        public long next()
        {
            z = (a * z + c) % m;
            return z;
        }

        public double nextDouble()
        {
            return next() / (double)m;
        }
    }
}
